src/library/scala/annotation/implicitNotFound.scala

```diff
@@ -52,5 +52,8 @@ package scala.annotation
  *   k.n[String]
  *      ^
  * </pre>
+ *
+ * @constructor Creates a new instance of the annotation with the provided message.
+ * @param msg The custom error message, must be a literal value.
```

Comment:
Changing the annotation to `extends ConstantAnnotation` would enforce that. It's not binary compatible, but since it's an annotation that doesn't matter. So IMO we could go ahead with that change.

Reply:
@lrytz so if I am understanding your comment right, you propose than instead of adding this comment, I should just make it `extend` **ConstantAnnotation** in this same PR?
Also, in that case, would you prefer if I also leave the extra documentation, or will it become redundant?

Reply:
Yes, I think this would be the better way forward. The `@param` message is still useful, the `@constructor` note seems not to add much value (?).

Reply:
> Yes, I think this would be the better way forward

Ok, I will make the change.

>  the @constructor note seems not to add much value (?).

I wasn't sure if you could add a `@param` annotation without a `@constructor` one.

Reply:
@lrytz it seems that change will break a lot of code 😬

Reply:
Interesting, let me take a look!

Reply:
This will go better after https://github.com/scala/scala/pull/9379 :-)